components/ui/checkbox.tsx

```diff
@@ -12,13 +16,40 @@ interface CheckboxProps {
   iconSize?: number;
 }
 
-const AnimatedCheck = Animated.createAnimatedComponent(Check);
+interface AnimatedCheckProps {
+  value: boolean;
+  size: number;
+  className: string;
+  opacity: number;
+}
+
+const AnimatedCheck = Animated.createAnimatedComponent(
+  React.forwardRef(
+    (
+      { value, size, className, opacity, ...props }: AnimatedCheckProps,
+      ref: React.ForwardedRef<any>
+    ) => {
+      return (
+        <View ref={ref} style={{ opacity }}>
```

Comment:
This kind of pattern makes it hard for reusables. I would include this in the checkbox component. It wouldn't need to be its own component.

Reply:
I was following the example here, where they made the animated components separately: https://docs.swmansion.com/react-native-reanimated/docs/core/createAnimatedComponent/#example

I did not plan to export the separate components from the file, so they'd be private in the file. If you'd prefer they're inside of the closure of the checkbox that's fine of course I'll update this with your code.